src/openrct2-ui/UiContext.cpp

```diff
@@ -103,6 +104,24 @@ class UiContext final : public IUiContext
         {
             SDLException::Throw("SDL_Init(SDL_INIT_VIDEO)");
         }
+
+        if (gConfigGeneral.window_scale == -1)
+        {
+#ifndef __MINGW32__
+            float ddpi, hdpi, vdpi;
+            if (SDL_GetDisplayDPI(0, &ddpi, &hdpi, &vdpi))
```

Comment:
This should store its (return) error code, and pass it to [SDL_GetError](https://wiki.libsdl.org/SDL_GetError). That way it'll provide us with some useful error information, rather than just crashing with an unspecified uncaught exception.